File: src/main/java/cn/apisium/beelogin/api/BeeLoginApi.java
package cn.apisium.beelogin.api;

import java.nio.charset.StandardCharsets;
import java.util.UUID;

// TODO finish these stuffs, replace all stub methods to functional methods.
public class BeeLoginApi {

	public static boolean authed(String token) {
		return false;
	}

	public static boolean authed(UUID authedUUID, String authedName) {
		return false;
	}

	public static String getName(String token) {
		return "";
	}

	public static UUID getUUID(String token) {
		return UUID.nameUUIDFromBytes((token + ":APISIUM").getBytes(StandardCharsets.UTF_8));
	}
}

File: src/main/java/cn/apisium/beelogin/api/BeeLoginApi.kt
package cn.apisium.beelogin.api

import org.bukkit.Server
import java.util.UUID

abstract class BeeLoginApi(val config: Map<String, String>, val id: String, val server: Server) {
  val list = hashMapOf<String, GameProfile>()

  abstract fun join(token: String): GameProfile

  abstract fun quit(token: String)

  abstract fun find(token: String): GameProfile?

  abstract fun findByName(name: String): GameProfile?

  abstract fun findById(id: UUID): GameProfile?

  fun check(id: String): Boolean = list.containsKey(id)
}
